### DISCUSSION TITLE
OffScreen - JavascriptBindingSettings.JavascriptBindingApiEnabled can no longer be modified, settings must be changed before the underlying browser has been created

### DISCUSSION THREAD

**Discussion by FurkanGozukara** [cefsharp/CefSharp#4333] [Q&A]
I am trying like below but it fails. I tried various cases all failed

`JavascriptBindingSettings.JavascriptBindingApiEnabled can no longer be modified, settings must be changed before the underlying browser has been created`

```
            var browser = new ChromiumWebBrowser(testUrl);

            //Disable creation of window.cefSharp and window.CefSharp
            browser.JavascriptObjectRepository.Settings.JavascriptBindingApiEnabled = false;

            //Rename object that CefSharp creates from cefSharp to notCefSharp (use your own name if you choose this option)
            browser.JavascriptObjectRepository.Settings.JavascriptBindingApiGlobalObjectName = "SomeNewBrowser";
```

.Net Core 7, CefSharp.OffScreen

**Comment by Rand0mRange** [comment]
Browser.JavascriptObjectRepository.Settings.LegacyBindingEnabled = true;
Browser.JavascriptObjectRepository.Register("CallbackService", new CallbackService());

Create a class in which you will handle calls from Javascript code, in this case - CallbackService.
You can call it whatever you like, I just gave an example.

**Comment by FurkanGozukara** [comment]
> browser.JavascriptObjectRepository.Settings.JavascriptBindingApiEnabled = false;
> 
>             //Rename object that CefSharp creates from cefSharp to notCefSharp (use your own name if you choose this option)
>             browser.JavascriptObjectRepository.Settings.JavascriptBindingApiGlobalObjectName = "SomeNewBrowser";

what do we write inside this call back service please elaborate the answer with more details 

also when do we set this. because after you init your browser if you call this it still throws error

Browser.JavascriptObjectRepository.Settings.LegacyBindingEnabled = true;

**Comment by FurkanGozukara** [comment]
@Rand0mRange I hate when someone doesn't know the answer but still answer in a  very vague way

found the answer from integration tests and it was so easy

```
            var browser = new ChromiumWebBrowser(testUrl, automaticallyCreateBrowser: false);

            browser.JavascriptObjectRepository.Settings.JavascriptBindingApiEnabled = false;

            browser.JavascriptObjectRepository.Settings.JavascriptBindingApiGlobalObjectName = "SomeNewBrowser";

            browser.CreateBrowser();
```

**Reply by amaitland** [reply]
> If you use [LegacyBindingEnabled](https://cefsharp.github.io/api/107.1.x/html/P_CefSharp_JavascriptBinding_JavascriptBindingSettings_LegacyBindingEnabled.htm) = true then you must set automaticallyCreateBrowser to false and call [CreateBrowser(IWindowInfo, IBrowserSettings)](https://cefsharp.github.io/api/107.1.x/html/M_CefSharp_OffScreen_ChromiumWebBrowser_CreateBrowser.htm) after the objects are registered.

As per https://cefsharp.github.io/api/107.1.x/html/T_CefSharp_OffScreen_ChromiumWebBrowser.htm
